client/src/components/Button.tsx

```diff
@@ -0,0 +1,26 @@
+import { ButtonProps, Button as BootstrapButton } from "react-bootstrap";
+import { useNavigate } from "react-router-dom";
+import React from "react";
+
+// Button wrapper for react-router
```

Comment:
decorator pattern 🤨?

Reply:
Kinda, if you think of components as functions. Traditionally they were classes, so it's like composition.

To apply the decorator pattern people use HOC (higher order components). This way you can add functionality to any element, rather than modifying a one-off. We might turn this into a HOC later on if there are more components that do server-side redirects.